extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/SqlStatementResourceHelper.java

```diff
@@ -137,6 +164,66 @@ public static long getLastIndex(Long numberOfRows, long start)
     return last;
   }
 
+  public static Optional<List<PageInformation>> populatePageList(
```

Comment:
I think this function could use a javadoc on how the page information is gathered from the counters.